Apps/WebClient/src/NewClientApp/src/stores/hospitalVisits.ts

```diff
@@ -0,0 +1,193 @@
+﻿import { defineStore } from "pinia";
+import { ref } from "vue";
+import { HospitalVisitState } from "@/models/datasetState";
+import { LoadStatus } from "@/models/storeOperations";
+import HospitalVisitResult from "@/models/hospitalVisitResult";
+import RequestResult from "@/models/requestResult";
+import { SERVICE_IDENTIFIER } from "@/ioc/identifier";
+import { container } from "@/ioc/container";
+import { IHospitalVisitService, ILogger } from "@/services/interfaces";
+import { HospitalVisit } from "@/models/encounter";
+import { ResultType } from "@/constants/resulttype";
+import { DatasetMapUtils } from "@/stores/utils/DatasetMapUtils";
+import EventTracker from "@/utility/eventTracker";
+import { EntryType } from "@/constants/entryType";
+import { ActionType } from "@/constants/actionType";
+import { ResultError } from "@/models/errors";
+import { ErrorSourceType, ErrorType } from "@/constants/errorType";
+import { useErrorStore } from "@/stores/error";
+
+const defaultHospitalVisitState: HospitalVisitState = {
+    data: [],
+    status: LoadStatus.NONE,
+    statusMessage: "",
+    error: undefined,
+    queued: false,
+};
+
+export const useHospitalVisitsStore = defineStore("hospitalVisits", () => {
+    const logger = container.get<ILogger>(SERVICE_IDENTIFIER.Logger);
+    const hospitalVisitService = container.get<IHospitalVisitService>(
+        SERVICE_IDENTIFIER.HospitalVisitService
+    );
+    const datasetMapUtil = new DatasetMapUtils<
+        HospitalVisit[],
+        HospitalVisitState
+    >(defaultHospitalVisitState);
+
+    const errorStore = useErrorStore();
+
+    const hospitalVisitsMap = ref(new Map<string, HospitalVisitState>());
+
+    function getHospitalVisitsState(hdid: string): HospitalVisitState {
+        return datasetMapUtil.getDatasetState(
+            hospitalVisitsMap.value,
+            hdid
+        ) as HospitalVisitState;
```

Comment:
```suggestion
        );
```
Type assertion should not be needed. (applies throughout)